network/src/network.rs

```diff
@@ -341,12 +298,14 @@ impl NetworkState {
             .collect::<Vec<_>>()
     }
 
-    pub(crate) fn can_dial(
-        &self,
-        peer_id: &PeerId,
-        addr: &Multiaddr,
-        allow_dial_to_self: bool,
-    ) -> bool {
+    pub(crate) fn can_dial(&self, addr: &Multiaddr, allow_dial_to_self: bool) -> bool {
+        let peer_id = extract_peer_id(addr);
+        if peer_id.is_none() {
+            trace!("Do not dial addr without peer id");
```

Comment:
Theoretically, we should never dial an addr without peer_id, it is better to use `error!` to logging this behaviour.